File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/Lab1/Repository/IRepository.cs
using System.Collections.Generic;
using Lab1.Domain;

namespace Lab1.Repository {
    public interface IRepository<ID, E> where E : Entity<ID> {
        E FindOne(ID id);
        List<E> FindAll();
        void Save(E entity);
        void Delete(ID id);
        void Update(E entity);
    }
}

File: 2nd Year/2nd Semester/MPP/Triatlon-Java/src/main/java/com/example/laborator/Repository/IRepository.java
package com.example.laborator.Repository;

import com.example.laborator.Domain.Entity;

public interface IRepository<ID, E extends Entity<ID>> {
    E findOne(ID id);

    Iterable<E> findAll();

    void save(E entity);

    void delete(ID id);

    void update(E entity);

}